### DISCUSSION TITLE
Homebrew Installation

### DISCUSSION THREAD

**Discussion by StephensMane** [Homebrew/discussions#2313] [Getting started]
Hello,

I Installed Homebrew on my macbook pro m1 yesterday. Right after I tried to install CoreUtils but the command brew install could not be found. I then run the homebrew install command a second time just to check if the installation was successful and if I missed anything in the installation process (I don't know if running the installation command of homebrew twice has actually installed it twice on my computer ?) which I did. After properly setting up HomeBrew I installed CoreUtils and I was able to use it right away. But this morning I'm not able to use the packages inside CoreUtils. Is that normal? Is there a line of command I gotta run to access the packages every time i open a new terminal?

Thank You for your help,

**Comment by carlocab** [comment]
Did you follow the instructions shown to you when the installer finished? It should've looked something like
```
- Add Homebrew to your PATH in ~/.zprofile:
    echo 'eval $(/opt/homebrew/bin/brew shellenv)' >> ~/.zprofile
    eval $(/opt/homebrew/bin/brew shellenv)
```

**Reply by StephensMane** [reply]
Hello,

Yes I missed it the first time but after I ran the install command the second time I read everything and figured it out ^^. Right after I installed CoreUtils and it all worked. I was able to use some command's options that are not available on mac os. But this morning as I return to my work I can't use coreutils. Homebrew is still installed I checked using some brew commands. So the problem is with coreutils definetely...

**Reply by carlocab** [reply]
What's the output of `brew info coreutils`, `brew config`, and `brew doctor`?

**Reply by StephensMane** [reply]
<img width="804" alt="Screenshot 2021-10-20 at 12 11 40" src="https://user-images.githubusercontent.com/60186600/138074449-a57d3af8-5c0c-473d-ade5-39b0d42a129b.png">

**Reply by carlocab** [reply]
Please [don't use screenshots](https://meta.stackoverflow.com/a/285557) when sharing the output of commands in your terminal. They're often difficult to read and impossible to reuse. Instead, copy the text here and wrap them in three back-ticks (```) on the line before and on the line after to format them properly.

What commands from `coreutils` did you try that don't work? Have you read the `caveats`?
```
==> Caveats
Commands also provided by macOS and the commands dir, dircolors, vdir have been installed with the prefix "g".
If you need to use these commands with their normal names, you can add a "gnubin" directory to your PATH with:
  PATH="/usr/local/opt/coreutils/libexec/gnubin:$PATH"
```

This should've been shown to you after doing `brew install coreutils`.

**Reply by StephensMane** [reply]
Sorry here is the output  

```
sh-3.2$ brew info coreutils
coreutils: stable 9.0 (bottled), HEAD
GNU File, Shell, and Text utilities
https://www.gnu.org/software/coreutils
Conflicts with:
  aardvark_shell_utils (because both install `realpath` binaries)
  b2sum (because both install `b2sum` binaries)
  ganglia (because both install `gstat` binaries)
  gdu (because both install `gdu` binaries)
  gegl (because both install `gcut` binaries)
  idutils (because both install `gid` and `gid.1`)
  md5sha1sum (because both install `md5sum` and `sha1sum` binaries)
  truncate (because both install `truncate` binaries)
  uutils-coreutils (because coreutils and uutils-coreutils install the same binaries)
/opt/homebrew/Cellar/coreutils/9.0 (480 files, 12.4MB) *
  Poured from bottle on 2021-10-19 at 21:02:25
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/coreutils.rb
License: GPL-3.0-or-later
==> Dependencies
Required: gmp ✔
==> Options
--HEAD
	Install HEAD version
==> Caveats
Commands also provided by macOS and the commands dir, dircolors, vdir have been installed with the prefix "g".
If you need to use these commands with their normal names, you can add a "gnubin" directory to your PATH with:
  PATH="/opt/homebrew/opt/coreutils/libexec/gnubin:$PATH"
==> Analytics
install: 147,148 (30 days), 264,689 (90 days), 766,633 (365 days)
install-on-request: 106,650 (30 days), 186,281 (90 days), 585,939 (365 days)
build-error: 0 (30 days)
```
```
sh-3.2$ brew config
HOMEBREW_VERSION: 3.2.17
ORIGIN: https://github.com/Homebrew/brew
HEAD: 556b8f16c6f8518404330780d153fd245b801325
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d894169005f6e50f16bc658756dde2bc10140c4d
Core tap last commit: 16 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.6-arm64
CLT: 13.0.0.0.1.1630607135
Xcode: N/A
Rosetta 2: false 
```
```
sh-3.2$ brew doctor
Your system is ready to brew. 
```

I did see that and I ran the command last night.

For example, I'm using the --complement option of cut in my script. This option can't be used on mac os but I was able to use it after installing coreutils. But not anymore.

Also in the output of brew info coreutils I can see "Conflicts with: " maybe there is the problem ?

**Reply by SMillerDev** [reply]
Did you add that snippet to your shell profile? Or did you only run it?

**Reply by carlocab** [reply]
Use `gcut` if you want `coreutils`' `cut` instead of the one that comes with macOS.

**Reply by StephensMane** [reply]
> Use `gcut` if you want `coreutils`' `cut` instead of the one that comes with macOS.

I ran the command 
```
PATH="/opt/homebrew/opt/coreutils/libexec/gnubin:$PATH"
```
and my script works again thank you. But I did ran this command last night and I thought it was one time for all. I will use 'gcut' from now thank you.

About this part 
```
Conflicts with:
  aardvark_shell_utils (because both install `realpath` binaries)
  b2sum (because both install `b2sum` binaries)
  ganglia (because both install `gstat` binaries)
  gdu (because both install `gdu` binaries)
  gegl (because both install `gcut` binaries)
  idutils (because both install `gid` and `gid.1`)
  md5sha1sum (because both install `md5sum` and `sha1sum` binaries)
  truncate (because both install `truncate` binaries)
  uutils-coreutils (because coreutils and uutils-coreutils install the same binaries)
```
is that normal or I may have something to do to fix this?

**Reply by StephensMane** [reply]
> Did you add that snippet to your shell profile? Or did you only run it?

Hello, thank you for helping. I'm not sure I understand. I'm new to bash, I'm a computer science student.

**Reply by carlocab** [reply]
Doing
```
PATH="/opt/homebrew/opt/coreutils/libexec/gnubin:$PATH"
```
only affects your `PATH` for your current session. If you want this to be permanent, you need to make a similar change [1] to your startup scripts [2].

[1] You probably want to `export` this.
[2] If you're using `zsh`, this is typically `.zprofile` or `.zshrc`. If you're using `bash`, then this is probably `.bashrc` or `.bash_profile`, or `.profile`.

**Reply by StephensMane** [reply]
> Doing
> 
> ```
> PATH="/opt/homebrew/opt/coreutils/libexec/gnubin:$PATH"
> ```
> 
> only affects your `PATH` for your current session. If you want this to be permanent, you need to make a similar change [1] to your startup scripts [2].
> 
> [1] You probably want to `export` this. [2] If you're using `zsh`, this is typically `.zprofile` or `.zshrc`. If you're using `bash`, then this is probably `.bashrc` or `.bash_profile`, or `.profile`.

Thanks ! I will fix this ! 👍
